torch/distributed/algorithms/ddp_comm_hooks/optimizer_overlap_hooks.py

```diff
@@ -21,9 +22,20 @@ def __init__(
             **functional_optim_kwargs,
             _allow_empty_param_list=True,
         )
-        if not hasattr(self.functional_optimizer, "step_param"):
+        self._check_valid_functional_optim()
+
+    @classmethod
+    def from_functional_optim(cls, functional_optim):
+        opt_hook_state_inst = cls.__new__(cls)  # Does not call __init__
+        opt_hook_state_inst.functional_optimizer = functional_optim
```

Comment:
(update: saw your unit test below that sets `_allow_empty_param_list` externally, please see my comment for that as well)

Should we set `_allow_empty_param_list` somewhere similar to `__init__`?

Reply:
Responded on the unittest comment!